pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/JuniperShipKitRoute.java

```diff
@@ -0,0 +1,268 @@
+import org.broadinstitute.dsm.util.KitUtil;
+import org.broadinstitute.dsm.util.externalshipper.ExternalShipper;
+import org.broadinstitute.lddp.handlers.util.Result;
+import org.broadinstitute.lddp.util.DeliveryAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+public class JuniperShipKitRoute extends RequestHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(JuniperShipKitRoute.class);
+
+    @Override
+    protected Object processRequest(Request request, Response response, String userId) throws Exception {
+        return null;
+    }
+
+    public Object createNonPepperKit(KitRequest kitUploadObject, String studyGuid, String kitTypeName,
+                                    boolean skipAddressValidation, String userIdRequest, AtomicReference<String> shippingCarrier) {
+        DDPInstanceDto ddpInstanceDto = new DDPInstanceDao().getDDPInstanceByStudyGuid(studyGuid).orElseThrow();
+        StringBuilder specialMessage = new StringBuilder();
+
+        HashMap<String, KitType> kitTypes = KitType.getKitLookup();
+        String key = kitTypeName + "_" + ddpInstanceDto.getDdpInstanceId();
+        KitType kitType = kitTypes.get(key);
+        if (kitType == null) {
+            throw new RuntimeException("KitType unknown");
+        }
+
+        Map<Integer, KitRequestSettings> kitRequestSettingsMap =
+                KitRequestSettings.getKitRequestSettings(String.valueOf(ddpInstanceDto.getDdpInstanceId()));
+        KitRequestSettings kitRequestSettings = kitRequestSettingsMap.get(kitType.getKitTypeId());
+        // if the kit type has sub kits > like for testBoston
+//        boolean kitHasSubKits = kitRequestSettings.getHasSubKits() != 0;
+
+        logger.info("Setup EasyPost...");
+        EasyPostUtil easyPostUtil = new EasyPostUtil(ddpInstanceDto.getInstanceName());
+
+        Map<String, KitRequest> invalidAddressList =
+                checkAddress(kitUploadObject, kitRequestSettings.getPhone(), skipAddressValidation, easyPostUtil);
+
+        //TODO Pegah discuss with Juniper if they need this, and if so, how the response should look like
+        List<KitRequest> duplicateKitList = new ArrayList<>();
+        ArrayList<KitRequest> orderKits = new ArrayList<>();
+
+        //TODO Pegah fix uploadKit to accept DDPInstanceDto
+        //getting the instance with isHasRole being set to true if the instance has role juniper_study
+        DDPInstance ddpInstance = DDPInstance.getDDPInstanceWithRole(ddpInstanceDto.getInstanceName(), "juniper_study");
+
+        if(!ddpInstance.isHasRole()){
+            throw new RuntimeException("This is not a Juniper study!");
+        }
+
+        TransactionWrapper.inTransaction(conn -> {
+            createKit(ddpInstance, kitType, kitUploadObject, kitRequestSettings, easyPostUtil, userIdRequest,
+                    kitTypeName, true, invalidAddressList, duplicateKitList, orderKits,
```

Comment:
For this code, and the methods below, is there corresponding code for Pepper?
IOW, don't we have to do something just like this for Pepper, or is there something unique about this code?

Reply:
I now made a method a lot simpler, tuned it to Juniper variables and also combined some methods all together